code/modules/events/vamp_teg.dm

```diff
@@ -0,0 +1,485 @@
+		if (!isnum(warning_delay) || warning_delay < 1)
+			return
+		var/event_duration = input(usr,"Duration of event? (Minutes)",src.name, 9) as num|null
+		if (!isnum(event_duration) || event_duration < 1)
+			return
+		var/grump_to_overcome = input(usr,"Grump to overcome?",src.name, 100) as num|null
+		if (!isnum(grump_to_overcome) || grump_to_overcome < 1)
+			return
+
+		src.event_effect(warning_delay, event_duration, grump_to_overcome)
+		return
+
+	event_effect(warning_delay, event_duration, grump_to_overcome)
+		..()
+		var/list/spooky_sounds = list("sound/ambience/nature/Wind_Cold1.ogg", "sound/ambience/nature/Wind_Cold2.ogg", "sound/ambience/nature/Wind_Cold3.ogg","sound/ambience/nature/Cave_Bugs.ogg", "sound/ambience/nature/Glacier_DeepRumbling1.ogg", "sound/effects/bones_break.ogg",	"sound/effects/gust.ogg", "sound/effects/static_horror.ogg", "sound/effects/blood.ogg")
+		var/list/area/stationAreas = get_accessible_station_areas()
+
+		if(!generator)
+			generator = locate(/obj/machinery/power/generatorTemp) in machine_registry[MACHINES_POWER]
+		if (!generator || generator.disposed || generator.z != Z_LEVEL_STATION )
+			message_admins("The Vampire TEG event failed to find TEG!")
+			return
+
+		if (!isnum(warning_delay))
+			warning_delay = rand(10, 50)
+		warning_delay = warning_delay SECONDS
+
+		if (!isnum(event_duration))
+			event_duration = rand(7, 9)
+		event_duration = event_duration MINUTES
+
+		if (!isnum(grump_to_overcome))
+			grump_to_overcome = 100
+
+		pda_connection = radio_controller.return_frequency("1149")
+
+		var/list/obj/machinery/station_switches = list()
+		for(var/area_key as() in stationAreas)
+			var/obj/machinery/light_switch/S
+			var/area/SA = stationAreas[area_key]
+			S = locate(/obj/machinery/light_switch) in SA?.machines
+			if(S)
+				station_switches += S
+
+		event_active = TRUE
+		target_grump =  max(generator.grump-50, 50)
+		generator.grump += grump_to_overcome
+		var/datum/effects/system/harmless_smoke_spread/smoke = new /datum/effects/system/harmless_smoke_spread()
+		smoke.set_up(1, 0, generator.loc)
+		smoke.attach(generator)
+		smoke.start()
+		playsound(generator, pick(spooky_sounds), 30, 0, -1)
+
+		src.circulators_to_relube = list(generator.circ1, generator.circ2)
+		for(var/obj/machinery/atmospherics/binary/circulatorTemp/C in circulators_to_relube)
+			C.reagents.add_reagent("black_goop", 10)
+			C.reagents.add_reagent("black_goop", 10)
+
+		// Delayed Warning and Instruction
+		SPAWN_DBG(warning_delay)
+			if(event_active)
+				command_alert("Reports indicate that the engine on-board [station_name()] is behaving unusually. Stationwide power failures may occur or worse.", "Engine Warning")
+				sleep(30 SECONDS)
+			if(event_active)
+				command_alert("Onsite Engineers inform us a sympathetic connection exists between the furances and the engine. Considering burning something it might enjoy food, people, weed, we are grasping at straws here. ", "Engine Suggestion")
```

Comment:
```suggestion
				command_alert("Onsite Engineers inform us a sympathetic connection exists between the furances and the engine. Considering burning something it might enjoy: food, people, weed. We're grasping at straws here. ", "Engine Suggestion")
```